lib/items.js

```diff
@@ -0,0 +1,82 @@
+/*!
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+const jsonmergepatch = require("json-merge-patch"),
+      joi = require("joi"),
+      UUID = require("uuid");
+      
+const DataStoreError = require("./util/errors");
+
+const STRING_500 = joi.string().max(500).allow("").allow(null).default("");
+const STRING_10K = joi.string().max(10000).allow("").allow(null).default("");
+const BASE_ENTRY_SCHEMA = joi.object().keys({
+  kind: joi.string().required(),
+  notes: STRING_10K
+});
+const ENTRY_SCHEMAS = [
+  BASE_ENTRY_SCHEMA.keys({
+    kind: "login",
+    username: STRING_500,
+    password: STRING_500
+  }),
+];
+const SCHEMA = joi.object().keys({
+  title: STRING_500,
+  origins: joi.array().items(STRING_500).max(5).default([]),
+  tags: joi.array().items(STRING_500).max(10).default([]),
```

Comment:
The max number of `origins` and `tags` seems a bit small. I'd expect these to be fairly large so that we're not keeping people from storing all the info they want. I thought the limits, at least in the datastore, were just sanity checks so no one screws up the DB too badly. (Though it might make sense to warn users in the UI if we think they're doing something weird.)

Reply:
I was arbitrary on my value selections for the array lengths.  However, my anecdotal observations make me think the limits fit what users would do.  I could be talked into more for `tags`, but `origins` carries real risks the larger it gets.